lib/logstash/inputs/http_poller.rb

```diff
@@ -91,77 +51,73 @@ def run(queue)
     end
   end
 
+  private
   def run_once(queue)
     @urls.each do |name, url|
-      @logger.debug? && @logger.debug("Will get url '#{name}' '#{url}")
-      client.async.get(url).
-        on_success {|response| handle_success(queue, name, url, response)}.
-        on_failure {|exception| handle_failure(queue, name, url, exception)}
+      request_async(queue, name, url)
+    end
+
+    # Some exceptions are only returned here! There is no callback :()
+    # For example, if there is a bad port number
+    client.execute!.each_with_index {|resp,i|
+      if resp.is_a?(java.lang.Exception) || resp.is_a?(StandardError)
+        name = @urls.keys[i]
+        url = @urls[name]
+        handle_failure(queue, name, url, resp)
       end
-    client.execute!
+    }
+  end
+
+  private
+  def request_async(queue, name, url)
+    @logger.debug? && @logger.debug("Will get url '#{name}' '#{url}")
+    client.async.get(url).
+      on_success {|response| handle_success(queue, name, url, response)}.
+      on_failure {|exception| handle_failure(queue, name, url, exception)}
   end
 
   private
   def handle_success(queue, name, url, response)
     @codec.decode(response.body) do |decoded|
-      event = LogStash::Event.new
-
-      event["name"] = name
-      event["host"] = @host
-      event["url"] = url
-      event["responseCode"] = response.code
-      event["message"] = decoded.to_hash
-      event["success"] = response.code >= 200 && response.code <= 299
-
-      queue << event
+      handle_decoded_event(queue, name, url, response, decoded)
     end
   end
 
   private
-  def handle_failure(queue, name, url, exception)
-    @logger.error("Cannot read URL! (#{exception}/#{exception.message})", :name => name, :url => url)
+  def handle_decoded_event(queue, name, url, response, event)
+    event["@metadata"] = event_metadata(name, url, response)
+    queue << event
+  rescue StandardError, java.lang.Exception => e
+    @logger.error? && @logger.error("Error eventifying response!", exception: e, name: name, url: url, response: response)
```

Comment:
We're still in the midst of style discussions, can we use the older hash-rocket syntax here instead until that's resolved?